lib/engine/game/g_1828.rb

```diff
@@ -268,7 +268,7 @@ def remove_extra_private_companies
                              .take(7 - @players.size)
         to_remove.each do |company|
           company.close!
-          @round.active_step.companies.delete(company)
+          @round.steps.find { |step| step.is_a?(Step::G1828::WaterfallAuction) }.companies.delete(company)
```

Comment:
should this be out of the hot loop?

Reply:
The loop is executed a maximum of four times, over two steps and only called during game setup, so I didn't optimize; however, if you want me to. I can cache the step. Let me know.

Reply:
Yeah, I try to avoid optimizations that won't have an impact on performance.